packages/opentelemetry-context-async-hooks/src/AbstractAsyncHooksContextManager.ts

```diff
@@ -177,6 +177,10 @@ implements ContextManager {
   ) {
     const contextManager = this;
     return function (this: never, event: string, listener: Func<void>) {
+      // do not double wrap. e.g. once calls on
+      if (contextManager._wrapped) {
```

Comment:
Even though i understand why this work i don't expect other to do so, i think a little comment explaining why its needed would be helpful WDYT ?

Reply:
Added a comment. Please take a look and make sure it makes sense.

Reply:
Looks good to me, lets see if other understand it